<!DOCTYPE html>
<html>
  <head>
    <title>Capstone project</title>
    <style>
      footer {
        background-color: #000;
        color: #fff;
        text-align: center;
        padding: 20px;
      }

      body {
        font-family: 'Times New Roman', Times, serif;
        margin: 0;
        padding: 0;
        background-color: #f2f2f2;
      }

      header {
        background-image: url("./background.jpg");
        color: black;
        padding: 20px;
        text-align: center;
        height: 100vh;
        display: flex;
        flex-direction: column;
        justify-content: center;
      }

      h1 {
        margin-top: 0;
        font-size: 3em;
        text-shadow: 2px 2px #fff;
      }

      nav {
        background-color: #333;
        color: #fff;
        padding: 10px;
      }

      nav a {
        color: #fff;
        margin: 0 10px;
        text-decoration: none;
        text-transform: uppercase;
      }

      @media screen and (max-width: 768px) {
        header {
          font-size: 24px;
        }

        nav {
          flex-direction: column;
        }

        nav a {
          margin: 10px 0;
        }
      }
    </style>
    <script>
      document.addEventListener("DOMContentLoaded", function () {
        // Get all the links in the navigation
        var navLinks = document.querySelectorAll("nav a");

        // Add click event listeners to each link
        navLinks.forEach(function (link) {
          link.addEventListener("click", function () {
            // Get the ID of the section to scroll to
            var sectionId = this.getAttribute("href");

            // Scroll to the section
            document.querySelector(sectionId).scrollIntoView({
              behavior: "smooth",
            });
          });
        });
      });
    </script>
  </head>
  <body>
    <nav>
      <a href="#ab">Abstract</a>
      <a href="#in">Intro</a>
      <a href="#oad">Objective</a>
      <a href="#data">Dataset</a>
      <a href="#met">Methodology</a>
      <a href="#imp">Implementation</a>
      <a href="#res">Results</a>
    </nav>

    <header>
      <h1>
        Discovering Potential Biomarkers for Uterine and Cervical Cancers With Machine Learning
      </h1>
      <h3>Advisor: Dr. Giri Narasimhan</h3>
      <h3>Mentor: Dr. Ananda Mondal</h3>
    </header>

    <section>
      <h2 id="ab" style="text-align: center">Abstract</h2>
      <p style="text-align: center">
Genetic information holds the power to give insight into the human condition- even revealing potential for illness in the future. Oncogenes, or cancer-causing genes, and tumor suppressing genes have been found to play a major role in the development of cancer. Though the scientific and medical communities are aware of several biomarkers for cervical and uterine cancers, this does not indicate the impossibility of others existing. In this study, three datasets of copy number alterations (CNA), gene expression, and microRNA (miRNA) were used, as well as one combined dataset consisting of overlapping participants in all three datasets. These data were used to find previously discovered biomarkers as well as new potential biomarkers via the feature importance of several different types of classification models. The important features were determined by SHapley Additive exPlanations, or SHAP values. The values were indicative of the overall impact of each feature for the model. 
Overall, it was found that among the three datasets, the most promising evidence was found in that of the miRNA,  HtSeq, and Combined datasets. One gene from the miRNA dataset, three genes from the HtSeq dataset, and four genes from the Combined dataset were found as potential candidates for biomarkers for uterine While these are currently biomarkers for only breast, gastric, & endocrine gland cancers, further research may be able to reveal their impact on the cancers evaluated in this study, uterine and cervical. 
      </p>

      <h2 id="in" style="text-align: center">Introduction</h2>
      <p style="text-align: center">
Though there have been a number of breakthroughs in recent years regarding cancer treatment, research must continue in order to find more effective methods of treatment and eventually a cure. While oncogenes were discovered over 50 years ago, they still remain a vital subject of interest in the search for better cancer treatments and prevention. A particular method of study involves the use of genetic information in order to determine a person’s predisposition for developing specific types of cancer. This predisposition for disease was coined as a ‘biomarker’. In machine learning, classification models can be used to represent the difference between classes in a dataset. By determining the feature importance of this genetic data, potential biomarkers for each  cancer are revealed.
      </p>

	<p style="text-align: center">
To properly explain all elements included, the introduction will be approached in a basic to detailed manner. This ensures that the reader, regardless of familiarity with the subject, will have learned something of value. First, the basic components of genetic material will be overviewed. Next, the types of genetic material used in this study will be described. Then, the types of cancer will be outlined. Finally, the aims, motivation, and significance of the work will be discussed. 
      </p>

	<p style="text-align: center">
To begin, some of the most basic components of genetics include nucleic acids named DNA and RNA. DNA is a molecule found in the nucleus of most cells that contains the genetic information for all of an organism’s functioning. A gene is a specific sequence of DNA that holds instructions to make some sort of molecule, whether that is a specific protein or an RNA molecule. RNA is another component that is quite essential for most biological functions. The process of producing a strand of RNA from DNA is called transcription. This RNA is either able to perform a certain function itself or forms instructions for the development of proteins, which is called translation. When an RNA molecule has instructions that do not strictly create proteins, the RNA would be called non-coding RNA. This sequence of events where DNA is transcribed to RNA and RNA creates protein is an integral part of biology called The Central Dogma.
      </p>

	<p style="text-align: center">
For the purpose of the study, a type of non-coding RNA called miRNA was used for analysis. These are small RNA that can alter gene expression, or the extent to which DNA is represented throughout the body, which will be discussed more thoroughly in the next paragraph. For the RNA that do help create proteins, they would have the title of messenger RNA (mRNA), as it carries the message of how to manufacture proteins. This process is called translation. Proteins are especially important because they help facilitate cellular functions that regulate the human body’s tissues and organs. Though there are other types of RNA, the ones that are most important to this study’s data types have already been discussed or fall under one of the two categories. 
      </p>
	<p style="text-align: center">
Copy number alterations (CNAs) occur when genes are altered in some way in a genome, which can result in adverse effects on cellular functioning. Copy number duplication is when a gene or a part of a gene is repeated and the copy number is increased, while copy number deletion is when a gene or a part of a gene is removed and the copy number is decreased. When a gene’s copy number is altered, the way that that gene is expressed is also changed. Gene expression refers to the way that the products of genetic information are converted into a product that carries out a certain function in the body. These are especially important in cancer, as over expressions and under expressions could result in specific genes being promoted and inhibited. When oncogenes are promoted, cancer cells are encouraged to continue growing throughout the body. Additionally, when there are tumor suppressor genes that are inhibited, the body has no defense against the growth of a tumor, thus resulting in uninhibited cancer cell growth. This can cause several complications in functions at the cellular level or, depending on the scale of the growth, across tissues and organs. For this reason, finding the specific genes that control these processes becomes especially important to successfully predict and hopefully prevent cancer. 
      </p>
	<p style="text-align: center">
Cervical cancer and uterine cancer are two cancers that negatively impact the lives of women across the globe. In the US alone, there were 11542 new cases of cervical cancer and 4272 deaths from 2016 to 2020. For uterine cancer, 54744 people were diagnosed with and 11995 women died of uterine cancer within that same time frame. Learning more information about cancers affecting the female reproductive system could lead to a mitigation of death by early detection or total prevention via biomarkers. Because these cancers have high metastasis, or the spreading of cancer to another part of the body, rates with each other due to proximity, they were chosen as the subjects of study for this experiment.      </p>



      <h2 id="oad" style="text-align: center">Objective</h2>
      <p style="text-align: center"></p>

      <p style="text-align: center">
        PACKAGES USED
      </p>
      
      <p style="text-align: center">• pandas</p>
      <p style="text-align: center">• numpy</p>
      <p style="text-align: center">• matplotlib.pyplot</p>
      <p style="text-align: center">• seaborn</p>
      <p style="text-align: center">• tensorflow</p>
      <p style="text-align: center">• tensorflow.keras.layers</p>
      <p style="text-align: center">• tensorflow.keras.models</p>
      <p style="text-align: center">• scikit learn</p>

      <p style="text-align: center">
        EXPLAIN THAT EXCEL AND POWERPOINT WERE ALSO USED
      </p>
      <h2 id="data" style="text-align: center">Code</h2>

      <p style="text-align: center">
        <a
          href="https://github.com/gaf17"
          >https://github.com/gaf17</a
        >
      </p>

      <h2 id="data" style="text-align: center">Dataset</h2>
      <p style="text-align: center">
		insert dataset source and snippet here
      </p>

      

      <h2 id="met" style="text-align: center">Methodology</h2>
      <p style="text-align: center">
        <p  style="text-align: center"><b>Data Pre-processing:</b> Boruta Feature Selection Algorithm & Normalization of Data</p>
		<p  style="text-align: center"><b>. Machine Learning Analysis:</b> Autoencoder Model (Artificial Neural Network with unsupervised learning of Latent Representation’s).</p>
      </p>

      <h2 id="imp" style="text-align: center">Implementation</h2>
      <p style="text-align: center">
        insert implementation
      </p>
	  <h4  style="text-align: center">A. Boruta Feature selection Algorithm:</h4>
      <p style="text-align: center">
        asdfasfasdfas
      </p>
	  <h4  style="text-align: center">B. Normalization:</h4>
      <p style="text-align: center">
       asdfadsfasdfasd
	      <br><b>Equation:</b>
<p>
	<p  style="text-align: center">Z = (x-mean)/ (standard deviation).</p>

<p  style="text-align: center">• Z is the data point's Normalized value.</p>
<p  style="text-align: center">• x = the original value of the data point</p>
<p  style="text-align: center">• mean = the mean of the dataset</p>
<p  style="text-align: center">• standard deviation = the standard deviation of
the dataset.</p>
</p>
      </p>
	  
	  <h4  style="text-align: center">C. Merging of Datasets:</h4>
      <p style="text-align: center">
        merge asfasfasdf
      </p>
	  <h4  style="text-align: center">D. Machine Learning Analysis:</h4>
      <p style="text-align: center">
        asdfasdf
	      <br>

Flow Chart of project:
      </p>
	  <div style="display: flex; justify-content: center;"><img src="imp.jpg" alt="#img"/></div>
	  
      

      <h2 id="res" style="text-align: center">Results</h2>
      <p style="text-align: center">
        asdfasdf
      </p>

      <p style="text-align: center">
        <img src="res1.png"alt="res 1" />
      </p>
      <p style="text-align: center">asdfasdf</p>
      <p style="text-align: center">
        <img src="res2.png" width="800"  alt="res2" />
      </p>
      <p style="text-align: center">asdfadsf
      </p>
      <p style="text-align: center">
        <img src="res3.png" width="800"  alt="res3" />
      </p>

     

      <h2 id="res" style="text-align: center">Conclusions</h2>
      <p style="text-align: center">
       conclusion
      </p>
    </section>

    <footer>
      <p style="text-align: center">Contributed by</p>
      <p style="text-align: center">
        <b>Genevieve Ferguson</b>
      </p>
	  <p>Master's Student in Data Science and AI
		gferg020@fiu.edu
		</p>
      <p style="text-align: center">IDC 6940: Capstone Project</p>
      <p style="text-align: center">MS in Data Science</p>
      <p style="text-align: center">Florida International University</p>
    </footer>
  </body>
</html>
